Content.Server/Objectives/Components/StealConditionComponent.cs

```diff
@@ -22,6 +22,19 @@ public sealed partial class StealConditionComponent : Component
     [DataField]
     public bool VerifyMapExistence = true;
 
+    /// <summary>
+    /// If true, only consider targets on a specific map during assignment.
+    /// Can be disabled for objectives where targets may be off-station (e.g., nukies).
+    /// </summary>
+    [DataField]
+    public bool FilterByMap = true;
+
+    /// <summary>
+    /// If filtering by map, use the mind owner's current map instead of the default map.
+    /// </summary>
+    [DataField]
+    public bool UseOwnerMap = false;
```

Comment:
These 2 bools should instead be a single enum field with 3 states `AnyMap`, `DefaultMap`, `MindOwnerCurrentMap`.
That way it will be easier to add new options in the future, and it will make the code cleaner in the system avoiding nested ifs.